org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -306,28 +310,46 @@ private void updateMarkers(Map<IResource, List<Issue>> issuesPerResource, Analys
       if (failedFiles.contains(resource)) {
         continue;
       }
-      List<Issue> rawIssues = resourceEntry.getValue();
-      try {
-        if (resource instanceof IFile) {
+      if (resource instanceof IFile) {
+        List<Issue> rawIssues = resourceEntry.getValue();
+        IPath path = resource.getFullPath();
+        try {
+          textFileBufferManager.connect(path, LocationKind.IFILE, new NullProgressMonitor());
+          ITextFileBuffer textFileBuffer = textFileBufferManager.getTextFileBuffer(path, LocationKind.IFILE);
+          IDocument document = textFileBuffer.getDocument();
+
           IProject project = resource.getProject();
           String moduleKey = project.getName();
           SonarLintCorePlugin.getDefault().getModulePathManager().setModulePath(moduleKey, project.getLocation().toString());
 
           String relativePath = resource.getProjectRelativePath().toString();
-          trackLocalIssues(moduleKey, resource, relativePath, rawIssues);
+          trackLocalIssues(document, moduleKey, relativePath, rawIssues);
           trackServerIssues(moduleKey, resource, relativePath);
-        } else {
-          // TODO delete if this never happens, or else handle it better
-          throw new IllegalStateException("updateMarkers for not an IFile?");
+        } finally {
+          textFileBufferManager.disconnect(path, LocationKind.IFILE, new NullProgressMonitor());
         }
-      } catch (Exception e) {
-        SonarLintCorePlugin.getDefault().error("Unable to compute position of SonarLint marker on resource " + resource.getName(), e);
+      } else {
+        // TODO delete if this never happens, or else handle it better
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)